source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,278 @@
+
+To retrieve files from a GridFS bucket, call the ``find()`` method on
+the ``GridFSBucket`` instance. The method returns a ``FindCursor`` instance
+from which you can access the results.
+
+The following code example shows you how to retrieve and print file metadata
+from all your files in a GridFS bucket. Among the different ways that you can
+traverse the retrieved results from the ``FindCursor`` iterable, the example
+below uses the ``forEach()`` method to display the results:
+
+.. code-block:: javascript
+
+   const cursor = bucket.find({});
+   cursor.forEach(doc => console.log(doc));
+
+The ``find()`` method accepts various query specifications and can be
+combined with other methods such as ``sort()``, ``limit()``, and ``project()``.
+
+For more information on the classes and methods mentioned in this section,
+see the following resources:
+
+- :node-api-4.0:`find() API documentation <classes/gridfsbucket.html#find>`
+- :node-api-4.0:`FindCursor API documentation <classes/findcursor.html>`
+- :doc:`Cursor Fundamentals page </fundamentals/crud/read-operations/cursor>`
+- :doc:`Read Operations page </fundamentals/crud/read-operations/>`
+
+.. _gridfs-download-files:
+
+Download Files
+--------------
+
+You can download files from your MongoDB database using a download
+stream. ``GridFSBucket`` contains the ``openDownloadStreamByName()``
+method which creates a download stream. 
+
+The example below shows you how to download a file referenced
+by the file name, stored in the ``filename`` field, into your working
+directory. Alternatively, you can use the ``openDownloadStream()``
+method, which takes the ``_id`` field of a file as a parameter. Note
+that if there are multiple documents with the same ``filename`` value,
+GridFS will stream the most recent file with the given name (as
+determined by the ``uploadDate`` field).
+
+.. code-block:: javascript
+
+   bucket.openDownloadStreamByName('myFile').
+        pipe(fs.createWriteStream('./outputFile'));
+
+.. note::
+
+   The GridFS streaming API cannot load partial chunks. When a download
+   stream needs to pull a chunk from MongoDB, it pulls the entire chunk
+   into memory. The 255 kilobyte default chunk size is usually
+   sufficient, but you can reduce the chunk size to reduce memory
+   overhead.
+
+For more information on the ``openDownloadStreamByName()`` method, see
+its :node-api-4.0:`API documentation <classes/gridfsbucket.html#opendownloadstreambyname>`.
+
+.. _gridfs-rename-files:
+
+Rename Files
+------------
+
+You can update the name of a GridFS file in your bucket by calling the
+``rename()`` method. You must specify the file to rename by its ``_id`` field
+rather than its file name.
+
+.. note::
+
+   The ``rename()`` method only supports updating the name of one file at
+   a time. To rename multiple files, retrieve a list of files matching the
+   file name from the bucket, extract the file id values from the files you
```

Comment:
```suggestion
   file name from the bucket, extract the ``_id`` field from the files you
```
Since you reference the id as _id in the intro, you should keep referencing it as that to avoid any confusion to the reader.

Applies to all instances.

Reply:
accepted